src/DynamicData.Tests/List/TransformAsyncFixture.cs

```diff
@@ -35,90 +38,127 @@ public void Dispose()
+        var person = new Person(key, 50);
+
+        _source.Add(person);
+        _source.Remove(person);
+
+        _results.Messages.Count.Should().Be(2, "Should be 2 updates");
+        _results.Messages.Count.Should().Be(2, "Should be 2 updates");
+        _results.Messages[0].Adds.Should().Be(1, "Should be 80 addes");
+        _results.Messages[1].Removes.Should().Be(1, "Should be 80 removes");
+        _results.Data.Count.Should().Be(0, "Should be nothing cached");
+    }
+
+    [Fact]
+    public void Update()
+    {
+        const string key = "Adult1";
+        var newperson = new Person(key, 50);
+        var updated = new Person(key, 51);
+
+        _source.Add(newperson);
+        _source.Add(updated);
+
+        _results.Messages.Count.Should().Be(2, "Should be 2 updates");
+        _results.Messages[0].Adds.Should().Be(1, "Should be 1 adds");
+        _results.Messages[0].Replaced.Should().Be(0, "Should be 1 update");
+    }
+
+    [Fact]
+    public async Task BatchOfUniqueUpdates()
+    {
+        var people = Enumerable.Range(1, 100).Select(i => new Person("Name" + i, i)).ToArray();
+
+        _source.AddRange(people);
+
+        _results.Messages.Count.Should().Be(1, "Should be 1 updates");
+        _results.Messages[0].Adds.Should().Be(100, "Should return 100 adds");
+
+        var tasks = people.Select(_transformFactory);
+        var result = await Task.WhenAll(tasks);
+
+        var transformed = result.OrderBy(p => p.Age).ToArray();
+        _results.Data.Items.OrderBy(p => p.Age).Should().BeEquivalentTo(_results.Data.Items.OrderBy(p => p.Age), "Incorrect transform result");
+    }
+
+    [Fact]
+    public void SameKeyChanges()
+    {
+        var people = Enumerable.Range(1, 10).Select(i => new Person("Name", i)).ToArray();
+
+        _source.AddRange(people);
+
+        _results.Messages.Count.Should().Be(1, "Should be 1 updates");
+        _results.Messages[0].Adds.Should().Be(10, "Should return 10 adds");
+        _results.Data.Count.Should().Be(10, "Should result in 10 records");
+    }
+
+    [Fact]
+    public void Clear()
+    {
+        var people = Enumerable.Range(1, 100).Select(l => new Person("Name" + l, l)).ToArray();
+
+        _source.AddRange(people);
+        _source.Clear();
+
+        _results.Messages.Count.Should().Be(2, "Should be 2 updates");
+        _results.Messages[0].Adds.Should().Be(100, "Should be 80 addes");
+        _results.Messages[1].Removes.Should().Be(100, "Should be 80 removes");
+        _results.Data.Count.Should().Be(0, "Should be nothing cached");
+    }
+
+    [Fact]
+    public void TransformOnRefresh()
+    {
+        var items = Enumerable.Range(1, 100).Select(i => new Person("Person" + i, 1)).ToArray();
+
+        //result should only be true when all items are set to true
+        using var list = new SourceList<Person>();
+        using var results = list.Connect().AutoRefresh(p => p.Age)
+            .TransformAsync(async v => v, transformOnRefresh: true).AsAggregator();
```

Comment:
v => Task.FromResult(v) will fix this warning

Reply:
👍